File: backend/python-nlp-katas/app/routes/tracks.py
from fastapi import APIRouter

router = APIRouter()

TRACKS = [
    {
        "id": "python-nlp",
        "name": "Python NLP",
        "description": "Learn NLP from foundations to transformers using Python. "
        "Build intuition for how text becomes numbers and why each technique exists.",
        "status": "active",
    },
    {
        "id": "rust-nlp",
        "name": "Rust NLP",
        "description": "Learn NLP from foundations to transformers using Rust. "
        "Build intuition for how text becomes numbers with systems-level control.",
        "status": "active",
    },
]


@router.get("/tracks")
async def list_tracks():
    return TRACKS

File: backend/rust-nlp-katas/src/routes/tracks.rs
use axum::Json;

use crate::models::kata::Track;

pub async fn list_tracks() -> Json<Vec<Track>> {
    Json(vec![
        Track {
            id: "python-nlp".into(),
            name: "Python NLP".into(),
            description: "Learn NLP from foundations to transformers using Python. \
                Build intuition for how text becomes numbers and why each technique exists."
                .into(),
            status: "active".into(),
        },
        Track {
            id: "rust-nlp".into(),
            name: "Rust NLP".into(),
            description: "Learn NLP from foundations to transformers using Rust. \
                Build intuition for how text becomes numbers with systems-level control."
                .into(),
            status: "active".into(),
        },
    ])
}
